fmt/src/formatter.rs

```diff
@@ -1691,6 +1691,10 @@ impl<'a, W: Write> Visitor for Formatter<'a, W> {
                     return Ok(())
                 }
 
+                if fmt.config.contract_new_lines {
+                    write_chunk!(fmt, "\n")?;
```

Comment:
```suggestion
                    fmt.write_whitespace_separator(true)?;
```

Reply:
So I tried using the helper but noticed my tests failed and I believe it's because of the not beginning of new line check at the start of `write_whitespace_seperator()`.

```rust
if !self.is_beginning_of_line() {
            write!(self.buf(), "{}", if multiline { "\n" } else { " " })?;
        }
```
Should I stick with `write_chunk!(fmt, "\n")?;` or is there another helper?

Reply:
uh, hm. forgot about that. lgtm then